src/shared/clients/iamClient.ts

```diff
@@ -77,11 +77,20 @@ export class DefaultIamClient {
         return tokens[tokens.length - 1]
     }
 
-    public async listAttachedRolePolicies(arn: string): Promise<IAM.ListAttachedRolePoliciesResponse> {
+    public async listAttachedRolePolicies(arn: string): Promise<IAM.AttachedPolicy[]> {
         const client = await this.createSdkClient()
         const roleName = this.getFriendlyName(arn)
-        const response = await client.listAttachedRolePolicies({ RoleName: roleName }).promise()
 
-        return response
+        const requester = async (request: IAM.ListAttachedRolePoliciesRequest) =>
+            client.listAttachedRolePolicies(request).promise()
+
+        const collection = pageableToCollection(requester, { RoleName: roleName }, 'Marker', 'AttachedPolicies')
+            .flatten()
+            .filter(p => p !== undefined)
+            .map(p => p!)
```

Comment:
Tradeoff is that this defeats any callers that would want to do paging themselves. But if we don't expect `listAttachedRolePolicies` to ever have many pages, that may be acceptable in this case.

Reply:
Yea, I would think the policies attached to an IAM role wouldn't require many pages since each policy can list as many permissions as it wants.